cmake/pika_setup_p2300.cmake

```diff
@@ -15,8 +15,9 @@ if(PIKA_WITH_P2300_REFERENCE_IMPLEMENTATION AND NOT PIKA_FIND_PACKAGE)
 
   fetchcontent_declare(
     P2300
-    GIT_REPOSITORY https://github.com/brycelelbach/wg21_P2300_std_execution.git
-    GIT_TAG 9159e4925a9895a6959e01e8e0ed22db4199be1a
+    # TODO: Upstream
+    GIT_REPOSITORY https://github.com/msimberg/wg21_P2300_std_execution.git
+    GIT_TAG 5d2898f556121fa3743ffa0dd5514d246eac76a7
```

Comment:
Testing from my branch until the change has been merged upstream. Change this to upstream before merging.

Reply:
Changed to upstream.